### PR TITLE
Fix LHN - Click and long pressing a user report with lots of messages opens popover inside report

### PR DESCRIPTION
**Pull request by carlosalmonte04**
### Details
Do not show the ReportActionContextMenu (popover menu) if OptionRowLHN (left hand sidebar link) has lost focus.

### Tests

1. While the left hand navigation rows are visible, long press one of the rows
2. Verify pop up shows
3. Dismiss the pop up by tapping outside
4. Tap on one of the items and quickly long press on the same item
5. Verify pop up does **not** show while on the report screen

- [x] Verify that no errors appear in the JS console

### Offline tests

Same steps as the Tests section above.

### QA Steps

Same steps as the Tests section above.

- [x] Verify that no errors appear in the JS console // no error related to the new code

### PR Author Checklist

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android / native
    - [x] Android / Chrome
    - [x] iOS / native
    - [x] iOS / Safari
    - [x] MacOS / Chrome / Safari
    - [x] MacOS / Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
    
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] If we are not using the full Onyx data that we loaded, I've added the proper selector in order to ensure the component only re-renders when the data it is using changes
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG)`)
    
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
    
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.

- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.

- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [x] I have checked off every checkbox in the PR author checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Web</summary>

https://github.com/carlosalmonte04/Expensify-App/assets/25206487/0d0d2694-53c3-49d9-afb2-b4d5872ae3c4

The right click context menu showing is an issue different from this one at 0:01, 0:06, etc....

</details>

<details>
<summary>Mobile Web - Chrome</summary>

https://github.com/carlosalmonte04/Expensify-App/assets/25206487/21b67e63-bd98-41ef-a39d-0946ab31ddc4

</details>

<details>
<summary>Mobile Web - Safari</summary>

https://github.com/carlosalmonte04/Expensify-App/assets/25206487/6914c28b-5b08-4aa2-b4a7-cd8c986ff0cc

</details>

<details>
<summary>Desktop</summary>

https://github.com/redpanda-bit/Expensify-App/assets/25206487/36dd9d42-2b66-403c-9741-e1237dce2c76

</details>

<details>
<summary>iOS</summary>

https://github.com/redpanda-bit/Expensify-App/assets/25206487/c3f4120a-341a-471c-827a-725958f2dcab

</details>

<details>
<summary>Android</summary>

https://github.com/redpanda-bit/Expensify-App/assets/25206487/14664c85-d383-481e-86bb-b6645c70c842

</details>

### CODE REVIEW

**Comment by redpanda-bit** [comment]
I have read the CLA Document and I hereby sign the CLA

**Comment by redpanda-bit** [comment]
recheck

File: src/components/LHNOptionsList/OptionRowLHN.js [hunk 1]
```diff
@@ -107,12 +109,26 @@ function OptionRowLHN(props) {
             ReportUtils.isWaitingForIOUActionFromCurrentUser(optionItem) ||
             (optionItem.isTaskReport && optionItem.isTaskAssignee && !optionItem.isCompletedTaskReport && !optionItem.isArchivedRoom));
 
+    useFocusEffect(
+        useCallback(() => {
+            isFocusedRef.current = true;
+            return () => {
+                if (!props.isSmallScreenWidth) {
```
Comment (sobitneupane) [inline src/components/LHNOptionsList/OptionRowLHN.js:116]: Instead of HOC, let's use `useWindowDimensions` hook to get `isSmallScreenWidth`.

Comment (redpanda-bit) [reply src/components/LHNOptionsList/OptionRowLHN.js]: Done.

File: src/components/LHNOptionsList/OptionRowLHN.js [hunk 2]
```diff
@@ -107,12 +111,26 @@ function OptionRowLHN(props) {
             ReportUtils.isWaitingForIOUActionFromCurrentUser(optionItem) ||
             (optionItem.isTaskReport && optionItem.isTaskAssignee && !optionItem.isCompletedTaskReport && !optionItem.isArchivedRoom));
 
+    useFocusEffect(
+        useCallback(() => {
+            isFocusedRef.current = true;
+            return () => {
+                if (!isSmallScreenWidth) {
+                    return;
+                }
```
Comment (sobitneupane) [inline src/components/LHNOptionsList/OptionRowLHN.js:120]:
```suggestion
```

File: src/components/LHNOptionsList/OptionRowLHN.js [hunk 3]
```diff
@@ -107,12 +111,26 @@ function OptionRowLHN(props) {
...
+            };
+        }, []),
+    );
     /**
      * Show the ReportActionContextMenu modal popover.
      *
      * @param {Object} [event] - A press event.
      */
     const showPopover = (event) => {
+        if (!isFocusedRef.current) {
```
Comment (sobitneupane) [inline src/components/LHNOptionsList/OptionRowLHN.js:131]:
```suggestion
        if (!isFocusedRef.current && isSmallScreenWidth) {
```

@redpanda-bit We can't use the memoized value returned by the useCallBack as `isSmallScreenWidth` is subject to change. The other way around might be to pass `isSmallScreenWidth` as dependency in the related useCallBack.

Comment (redpanda-bit) [reply src/components/LHNOptionsList/OptionRowLHN.js]: I totally missed that. Thanks for catching that. Made the changes and re-tested. Please let me know if you see something else.

**Review by sobitneupane — APPROVED** [review]
## Reviewer Checklist

- [x] I have verified the author checklist is complete (all boxes are checked off).
- [x] I verified the correct issue is linked in the `### Fixed Issues` section above
- [x] I verified testing steps are clear and they cover the changes made in this PR
    - [x] I verified the steps for local testing are in the `Tests` section
    - [x] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [x] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [x] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I verified tests pass on **all platforms** & I tested again on:
    - [x] Android / native
    - [x] Android / Chrome
    - [x] iOS / native
    - [x] iOS / Safari
    - [x] MacOS / Chrome / Safari
    - [x] MacOS / Desktop
- [x] If there are any errors in the console that are unrelated to this PR, I either fixed them (preferred) or linked to where I reported them in Slack
- [x] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` have been tested & I retested again)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [x] I have checked off every checkbox in the PR reviewer checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Web</summary>


https://github.com/Expensify/App/assets/25876548/c63c2fb4-8b63-4f10-bed4-dc635907c924


</details>

<details>
<summary>Mobile Web - Chrome</summary>

https://github.com/Expensify/App/assets/25876548/8fd8359b-254d-4f7d-814f-bf31ccd0ffd2


</details>

<details>
<summary>Mobile Web - Safari</summary>

https://github.com/Expensify/App/assets/25876548/6ebb7aa2-0ab6-489e-9c75-e2c08abe704d

</details>

<details>
<summary>Desktop</summary>



https://github.com/Expensify/App/assets/25876548/a3294258-589e-44ec-a258-13cc6c98237c


</details>

<details>
<summary>iOS</summary>

https://github.com/Expensify/App/assets/25876548/e2ace87b-2b3a-4a12-a36e-0b33fddc74de


</details>

<details>
<summary>Android</summary>

https://github.com/Expensify/App/assets/25876548/6c5de04c-406b-402b-a27b-6c68da1c2aab



</details>

**Review by Gonals — DISMISSED** [review]

**Comment by Gonals** [comment]
@redpanda-bit, seems like we have unsigned commits!

**Comment by redpanda-bit** [comment]
@Gonals sorry I thought I had configured my git properly. Would I need to force push now that I have the commits signed?

**Comment by Gonals** [comment]
> @Gonals sorry I thought I had configured my git properly. Would I need to force push now that I have the commits signed?

Give that a try, yes!

**Comment by redpanda-bit** [comment]
@Gonals done, thanks so much.

**Comment by Gonals** [comment]
There are a couple of lint issues left, regarding conditionally calling hooks!

**Comment by redpanda-bit** [comment]
No way! Thanks a lot for being patient, I'm used to jest and lint running on git hooks. 

1. Fixed the lint errors by moving hooks above conditional renders – there are other `rulesdir/no-multiple-onyx-in-file` lint errors but the rule was introduced recently, so I'm guessing someone is working on fixing those errors? I can take care of those if you'd like. Please let me know.
2. Merged main to fix the failing jest unit test.

File: src/components/LHNOptionsList/OptionRowLHN.js [hunk 4]
```diff
@@ -106,19 +115,8 @@ function OptionRowLHN(props) {
     const hasBrickError = optionItem.brickRoadIndicator === CONST.BRICK_ROAD_INDICATOR_STATUS.ERROR;
     const defaultSubscriptSize = optionItem.isExpenseRequest ? CONST.AVATAR_SIZE.SMALL_NORMAL : CONST.AVATAR_SIZE.DEFAULT;
     const shouldShowGreenDotIndicator =
-        !hasBrickError &&
-        (optionItem.isUnreadWithMention ||
-            ReportUtils.isWaitingForIOUActionFromCurrentUser(optionItem) ||
-            (optionItem.isTaskReport && optionItem.isTaskAssignee && !optionItem.isCompletedTaskReport && !optionItem.isArchivedRoom));
+        !hasBrickError && (optionItem.isUnreadWithMention || optionItem.isWaitingForTaskCompleteFromAssignee || ReportUtils.isWaitingForIOUActionFromCurrentUser(optionItem));
```
Comment (redpanda-bit) [inline src/components/LHNOptionsList/OptionRowLHN.js]: I did not introduced these changes! Is this normal on a merge workflow? This doesn't happen in rebase workflows.

File: src/components/LHNOptionsList/OptionRowLHN.js [hunk 5]
```diff
@@ -195,6 +193,7 @@ function OptionRowLHN(props) {
                         ]}
                         accessibilityRole={CONST.ACCESSIBILITY_ROLE.BUTTON}
                         accessibilityLabel={translate('accessibilityHints.navigatesToChat')}
+                        needsOffscreenAlphaCompositing={props.optionItem.icons.length >= 2}
```
Comment (redpanda-bit) [inline src/components/LHNOptionsList/OptionRowLHN.js]: Same here.

File: src/components/LHNOptionsList/OptionRowLHN.js (hunk 4 above)
Comment (Gonals) [reply src/components/LHNOptionsList/OptionRowLHN.js]: Ah, I imagine something got messes up with the re-push. I'd recommend just moving your changes to a new branch fresh from `main`

Comment (redpanda-bit) [reply src/components/LHNOptionsList/OptionRowLHN.js]: 😞 a lot of learnings, I tell you. Here is the new branch: https://github.com/redpanda-bit/Expensify-App/tree/fix/popover-showing.

Comment (Gonals) [reply src/components/LHNOptionsList/OptionRowLHN.js]: At this point, I think you can just create a PR for that branch and we'll close this one

Comment (redpanda-bit) [reply src/components/LHNOptionsList/OptionRowLHN.js]: Here is the new PR: https://github.com/Expensify/App/pull/28614.

**Comment by Gonals** [comment]
Moved to https://github.com/Expensify/App/pull/28614 due to various merge issues. Closing

**Pull request closed without merging**